jena-text/src/main/java/org/apache/jena/query/text/TextIndexLucene.java

```diff
@@ -399,4 +404,24 @@ private Node entryToNode(String v) {
         // TEMP
         return NodeFactoryExtra.createLiteralNode(v, null, null) ;
     }
+    
+    
+    // set up event handlers
+    
+    public enum Event { CLOSED };
+    
+    public static interface EventHandler {
+    	void handle(TextIndexLucene index);
+    }
+    
+    private Map<Event, List<EventHandler>> eventHandlers = new HashMap<Event, List<EventHandler>>();
+    
+    public void addEventHandler(Event event, EventHandler handler) {
+    	List<EventHandler> handlers = eventHandlers.get(event);
```

Comment:
I think this and the `if-then` below might just be `eventHandlers.computeIfAbsent(event, e -> new ArrayList<>())`?